play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -141,7 +142,23 @@ private[jackson] class JsValueDeserializer(factory: TypeFactory, klass: Class[_]
 
     val (maybeValue, nextContext) = (jp.getCurrentToken.id(): @switch) match {
 
-      case JsonTokenId.ID_NUMBER_INT | JsonTokenId.ID_NUMBER_FLOAT => (Some(JsNumber(jp.getDecimalValue)), parserContext)
+      case JsonTokenId.ID_NUMBER_INT | JsonTokenId.ID_NUMBER_FLOAT => {
+        val inputText = jp.getText
+        val inputLength = inputText.length
+
+        // There is a limit of how large the numbers can be since parsing extremely
+        // large numbers (think thousand of digits) and operating on the parsed values
+        // can potentially cause a DDoS.
+        if (inputLength > JacksonJson.BigDecimalLimits.DigitsLimit) throw new IllegalArgumentException("Number is larger than supported")
+
+        // Must create the BigDecimal with a MathContext that is consistent with the limits used.
+        val bigDecimal = BigDecimal(inputText, JacksonJson.BigDecimalLimits.DefaultMathContext)
+
+        // We should also avoid numbers with scale that are out of a safe limit
+        if (Math.abs(bigDecimal.scale) > JacksonJson.BigDecimalLimits.ScaleLimit) throw new IllegalArgumentException(s"Number scale (${bigDecimal.scale}) is out of limits")
+
+        (Some(JsNumber(bigDecimal)), parserContext)
+      }
```

Comment:
> (nitpick) a 15-line block breaks readability of the switch statement.

Solved.

> (nitpick) how does JacksonJson.BigDecimalLimits relate to serialiser limits? We are destructive in serilisation and relaxed in parsing (still accept huge values). The inconsistency already existed so we may want to improve in a separate PR.

Well, the serializer is not imposing any limits besides how to represent the number (using plain or scientific notation). My rationale here was that it is up to the user to validate if the number is within limits when serializing (considering large numbers) because the BigDecimal is already there, it already exists and was computed before serializing.

WDYT?

> (nitpick) IT'd be great if the exceptions included the name of the field containing the offending data. Something like val maybeFieldName = parserContext.find(_.isInstanceOf[KeyRead]).map{_.asInstanceOf[KeyRead].fieldName}? This is a security validation/restriction so I'm not 100% sure it's really necessary to add this extra info though.

Done.